cmd/docker/builder.go

```diff
@@ -73,6 +73,12 @@ func processBuilder(dockerCli command.Cli, cmd *cobra.Command, args, osargs []st
 		return args, osargs, nil
 	}
 
+	// wcow build command must use the legacy builder
+	// if not opt-in through a builder component
+	if !useBuilder && dockerCli.ServerInfo().OSType == "windows" {
```

Comment:
I'm slightly concerned that there's now 2 implementations to do the BuildKit check (one in `BuildKitEnabled()`); I don't want to necessarily block this PR on that, but wondering if we can use the same code for both (so that we're sure the logic won't diverge)

Reply:
Yes, I was thinking of moving some logic to `BuildKitEnabled` but in `processBuilder` we need to know if `DOCKER_BUILDKIT` is being used/enforced.

Reply:
could we extract the common bits to a non-exported function perhaps? (haven't checked), so something like;

```go
func (cli *DockerCli) BuildKitEnabled() (bool, error) {
    foo, _, err := doTheBuildKitCheck(cli)
    return foo, err
}
```

Reply:
hum yeah we could that, let me see.